File: java/parsers_java21/src/main/java/jp/co/teruuu/parser/common/SkipParser.java
package jp.co.teruuu.parser.common;

public class SkipParser implements Parser<Void> {
    private final String literal;
    public SkipParser(String literal) {
        this.literal = literal;
    }

    @Override
    public ParseResult<Void> parse(String input, int location) {
        if (input.startsWith(literal, location)) {
            return new ParseResult.Success<>(null, location + literal.length());
        } else {
            return new ParseResult.Failure<>(String.format("not (literal=[%s], loc=[%d]), input=%s", literal, location, input), location);
        }
    }
}

File: rust_docker/parser_back/src/core/util/skip_parser.rs
use std::rc::Rc;

use crate::core::parser::Parser;
use crate::core::parse_result::ParseResult;

pub fn skip_parser<'a, A, F>(parser: Parser<'a, A, F>) -> Parser<'a, (), impl Fn(&'a [u8], usize) -> ParseResult<()> + Clone + 'a>
where
    A: Clone + 'a,
    F: Fn(&'a [u8], usize) -> ParseResult<A> + 'a,
{
    let f1 = parser.parse.clone();

    let f = move |input: &'a [u8], location: usize| {
        match f1(input, location) {
            ParseResult::Success { value: _, location } => ParseResult::Success { value: (), location },
            ParseResult::Failure { parse_error, location } => ParseResult::Failure { parse_error, location }
        }
    };

    Parser::new(Rc::new(f))
}